website/server/models/challenge.js

```diff
@@ -149,7 +154,7 @@ schema.methods.syncToUser = async function syncChallengeToUser (user) {
     }
   });
 
-  toSave.push(user.save());
+  toSave.push(User.collection.update({_id: user._id}, user.toObject()));
```

Comment:
This is a lot of data to update on the user object. Why not just the things that are being added to it?